retail-managment-system/rms-ui/src/app/pages/customers/customers.component.ts

```diff
@@ -23,31 +24,96 @@ export class CustomersComponent implements OnInit {
     'secondaryPhoneNo',
     'address',
     'trustReceiptNo',
-    'update',
-    'delete',
+    'actions',
   ];
 
   constructor(
     private customersRepository: CustomersRepository,
-    public dialog: MatDialog
+    private build: FormBuilder
   ) {}
 
   ngOnInit() {
     this.getAllCustomers();
+    this.custForm();
   }
+
+  custForm() {
+    this.customersForm = this.build.group({
+      id: [''],
+      fullName: ['', [Validators.required]],
+      nickName: [''],
+      customerCode: [''],
+      version: [''],
+      nationalId: [
+        '',
+        [
+          Validators.required,
+          Validators.pattern('^[0-9]*$'),
+          Validators.minLength(14),
+          Validators.maxLength(14),
+        ],
+      ],
+      primaryPhoneNo: [
+        '',
+        [Validators.required, Validators.pattern('^01[0-2,5]{1}[0-9]{8}$')],
+      ],
+      secondaryPhoneNo: ['', [Validators.pattern('^01[0-2,5]{1}[0-9]{8}$')]],
+      address: ['', [Validators.required]],
+      trustReceiptNo: [''],
+    });
+  }
+
   getAllCustomers(): void {
     this.customersRepository.getList().subscribe((result) => {
       this.allCustomers = result;
     });
   }
-  openDialog(element: customers | null) {
-    this.dialog
-      .open(ManageCustomersComponent, {
-        data: element,
-      })
-      .afterClosed()
-      .subscribe(() => {
-        this.getAllCustomers();
-      });
+
+  fetchData(customer: Customers): void {
+    this.isButtonVisible = false;
+    this.customersForm.patchValue(customer);
+    this.currentData = customer;
+  }
+
+  updateCustomer() {
+    this.isButtonVisible = true;
+    this.submit = true;
+    this.customersRepository.update(this.customersForm.value).subscribe(() => {
+      this.getAllCustomers();
+      this.submit = false;
+    });
+  }
+
+  addCustomer() {
+    this.isButtonVisible = true;
+    this.submit = true;
+    this.customersRepository.add(this.customersForm.value).subscribe(() => {
+      this.getAllCustomers();
+      this.submit = false;
+    });
+  }
+
+  onSubmit() {
+    if (this.customersForm.valid) {
+      this.customersForm.controls['id'].value
+        ? this.updateCustomer()
+        : this.addCustomer();
+      this.customersForm.reset();
+    }
+  }
+  resetForm(): void {
+    this.customersForm.controls['id'].value
+      ? this.fetchData(this.currentData)
+      : this.customersForm.reset();
+  }
+
+  resetTheForm(): void {
```

Comment:
name must be more better 
what is different between this and reset function for anyone read code?